R/extract_diagnostics.R

```diff
@@ -0,0 +1,161 @@
+#' Extract diagnostic metrics from model fit and data
+#'
+#' This function extracts various diagnostic metrics from a fitted `epinow2`
+#' model and provided data. It checks for low case counts and computes
+#' diagnostics from the fitted model, including the mean acceptance
+#' statistic, divergent transitions, maximum tree depth, and Rhat values.
+#' These diagnostics are then flagged if they exceed specific thresholds,
+#' and the results are returned as a data frame.
+#'
+#' @param fit A list containing the model fit object from `epinow2`, which
+#'            includes `estimates$fit`.
+#' @param data A data frame containing the input data used in the model fit.
+#' @param job_id A unique identifier for the job or task being processed.
+#' @param task_id A unique identifier for the task being performed.
+#' @param disease,geo_value,model Metadata for downstream processing.
+#'
+#' @return A \code{data.frame} containing the extracted diagnostic metrics. The
+#' data frame includes the following columns:
+#' \itemize{
+#'   \item \code{diagnostic}: The name of the diagnostic metric.
+#'   \item \code{value}: The value of the diagnostic metric.
+#'   \item \code{state}: The state for which the model was run.
+#'   \item \code{disease}: The disease/pathogen being analyzed.
+#'   \item \code{job_id}: The unique identifier for the job.
+#'   \item \code{task_id}: The unique identifier for the task.
+#'   \item \code{disease,geo_value,model}: Metadata for downstream processing.
```

Comment:
disease is here again, as well as above? Why not have new lines for these and say what `geo_value` and `model` are?

Reply:
https://github.com/CDCgov/cfa-epinow2-pipeline/pull/41/commits/88b46800dbc647a5de78112c3f4e67b8a9218cce